controllers/tf_controller.go

```diff
@@ -136,7 +136,6 @@ func (r *TerraformReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 
 		// Record Prometheus metrics.
-		r.Metrics.RecordReadiness(ctx, &terraform)
 		r.Metrics.RecordSuspend(ctx, &terraform, terraform.Spec.Suspend)
 		r.Metrics.RecordDuration(ctx, &terraform, reconcileStart)
```

Comment:
Are these safe from this? They care using the same reference, or are those functions check extra criteria that wasn't checked in `RecordReadiness` so they are not affected by the same issue?

Reply:
Yes, these functions check extra criteria that aren't checked in the RecordReadiness, so they are not affected by the overwrite issue.